src/fu-tool.c

```diff
@@ -2781,6 +2784,48 @@ fu_util_switch_branch (FuUtilPrivate *priv, gchar **values, GError **error)
 	return fu_util_prompt_complete (priv->completion_flags, TRUE, error);
 }
 
+static gboolean
+fu_util_lock (FuUtilPrivate *priv, GError **error)
+{
+#ifdef HAVE_WRLCK
+	struct flock lockp = {
+		.l_type = F_WRLCK,
+		.l_whence = SEEK_SET,
+	};
+	g_autofree gchar *localstatedir = NULL;
+	g_autofree gchar *lockfn = NULL;
+
+	/* open file */
+	localstatedir = fu_common_get_path (FU_PATH_KIND_LOCALSTATEDIR);
+	lockfn = g_build_filename (localstatedir, "run", "fwuptool", NULL);
+	if (!fu_common_mkdir_parent (lockfn, error))
+		return FALSE;
+	priv->lock_fd = g_open (lockfn, O_RDWR | O_CREAT, S_IRWXU);
+	if (priv->lock_fd < 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "failed to open %s",
+			     lockfn);
+		return FALSE;
+	}
+
+	/* write lock */
+	if (fcntl (priv->lock_fd, F_OFD_SETLK, &lockp) < 0) {
```

Comment:
I suppose it's not actually required to set `F_UNLCK` on close because the descriptor releases, right?

Reply:
Yup, that's why I chose ofd rather than a file lock.

Reply:
Sorry, I should fully complete my thoughts - if the fwupdtool crashes I want to make sure that we don't end up with dangling descriptors, which I don't know if that's a thing.